File: core/application/src/main/java/com/gonezo/multiplatform/audioextraction/application/ResolutionCoordinator.java
package com.gonezo.multiplatform.audioextraction.application;

import com.gonezo.multiplatform.audioextraction.domain.model.ExtractionContext;
import com.gonezo.multiplatform.audioextraction.domain.model.FieldCandidate;
import com.gonezo.multiplatform.audioextraction.domain.model.ResolvedField;
import com.gonezo.multiplatform.audioextraction.domain.schema.OutputSchema;
import java.util.List;
import java.util.Map;

public interface ResolutionCoordinator {
  Map<String, ResolvedField> resolve(
    Map<String, List<FieldCandidate>> candidates,
    OutputSchema schema,
    ExtractionContext context
  );
}

File: core/bin/main/com/gonezo/audioextraction/application/pipeline/ResolutionCoordinator.kt
package com.gonezo.audioextraction.application.pipeline

import com.gonezo.audioextraction.domain.model.ExtractionContext
import com.gonezo.audioextraction.domain.model.FieldCandidate
import com.gonezo.audioextraction.domain.model.ResolvedField
import com.gonezo.audioextraction.domain.schema.OutputSchema

interface ResolutionCoordinator {
  fun resolve(
    candidates: Map<String, List<FieldCandidate>>,
    schema: OutputSchema,
    context: ExtractionContext,
  ): Map<String, ResolvedField>
}
